### PR TITLE
UITEST-111 Updated `<AdvancedSearch>` interactor for search match support. Updated e2e tests.

### PR DESCRIPTION
**Pull request by BogdanDenis**
## Description
Updated `advanced-search` interactor to support search match selector. Deleted duplicate `advanced-search-inventory` interactor as it has the same implementation, but with those additional match selector options.
Updated e2e tests to use `advanced-search` interactor instead of `advanced-search-inventory` interactor

## Issues
[UITEST-111](https://issues.folio.org/browse/UITEST-111)

## Related PRs
https://github.com/folio-org/stripes-components/pull/2141
https://github.com/folio-org/stripes-authority-components/pull/116

### CODE REVIEW

**Review by Dmytro-Melnyshyn — APPROVED** [review]

File: interactors/advanced-search.js [hunk 1]
```diff
@@ -3,17 +3,23 @@ import Select from './select';
...
+    option: el => el.querySelectorAll(`select#advanced-search-option-${getRowIndex(el)}`).value,
   })
   .actions({
     fillQuery: ({ find }, value) => find(TextArea({ ariaLabel: 'Search for' })).fillIn(value),
-    selectBoolean: ({ find }, rowIndex, value) => find(Select({ id: `advanced-search-bool-${rowIndex}` })).choose(value),
+    selectBoolean: ({ find }, rowIndex, value) => find(Select({ id: `advanced-search-bool-${rowIndex}` }))
+      .choose(value),
+    selectMatchOption: ({ find }, rowIndex, value) => find(Select({ id: `advanced-search-match-${rowIndex}` }))
+      .choose(value),
     selectSearchOption: ({ find }, rowIndex, value) => find(Select({ id: `advanced-search-option-${rowIndex}` })).choose(value),
```
Comment (zentestuken) [inline interactors/advanced-search.js:23]: Here, the actual id of match option select would be `advanced-match-${rowIndex}`

File: interactors/advanced-search.js [hunk 2]
```diff
@@ -3,17 +3,23 @@ import Select from './select';
...
   .selector('[class*=AdvancedSearchRow-]')
   .filters({
-    index: (el) => parseInt(el.getAttribute('data-row-index'), 10),
-    query: (el) => el.querySelector('textarea').getAttribute('value'),
-    bool: (el) => el.querySelectorAll('select')[0].value,
-    option: (el) => el.querySelectorAll('select')[1].value,
+    index: el => getRowIndex(el),
+    query: el => el.querySelector('textarea').getAttribute('value'),
+    bool: el => el.querySelectorAll(`select#advanced-search-bool-${getRowIndex(el)}`).value,
+    matchOption: el => el.querySelectorAll(`select#advanced-search-match-${getRowIndex(el)}`).value,
```
Comment (zentestuken) [inline interactors/advanced-search.js:14]: Here, the actual id of match option select would be `advanced-match-${rowIndex}`

File: interactors/advanced-search.js [hunk 3]
```diff
@@ -3,17 +3,23 @@ import Select from './select';
...
+    option: el => el.querySelectorAll(`select#advanced-search-option-${getRowIndex(el)}`).value,
   })
   .actions({
     fillQuery: ({ find }, value) => find(TextArea({ ariaLabel: 'Search for' })).fillIn(value),
-    selectBoolean: ({ find }, rowIndex, value) => find(Select({ id: `advanced-search-bool-${rowIndex}` })).choose(value),
+    selectBoolean: ({ find }, rowIndex, value) => find(Select({ id: `advanced-search-bool-${rowIndex}` }))
+      .choose(value),
+    selectMatchOption: ({ find }, rowIndex, value) => find(Select({ id: `advanced-search-match-${rowIndex}` }))
```
Comment (zentestuken) [inline interactors/advanced-search.js:21]: Here, the actual id of match option select would be `advanced-match-${rowIndex}`

**Review by zentestuken — CHANGES_REQUESTED** [review]

File: interactors/advanced-search.js (hunk 2 above)
Comment (BogdanDenis) [reply interactors/advanced-search.js:14]: I changed the ids in the components to be consistent with the rest
https://github.com/folio-org/stripes-components/pull/2141/files#diff-1b9916b8e438211b16da2942c780ae6f074fd363ef3e4dc03b7f32cda2e9c8e2R121

**Review by zentestuken — APPROVED** [review]

**Pull request merged**